### PR TITLE
ci: add integration-tests job to test-pull-request 

### PR DESCRIPTION
**Pull request by lavarou**
integration-tests job in test-pull-request workflow runs:
* all LASP tests
* all custom events limits tests
* all integration tests

This requires refactoring of the build system:
* Separation of agent service into nr-php and devenv services; nr-php service is used by integration-tests job, devenv service is local agent development;
* Make LASP and custom events limits integration tests less verbose.
* Make agent's integration tests less verbose. 
* Relax LASP and custom events limits integration tests dependency to
only integration_runner binary only and add a rule to build it.
* Break out integration tests into its own target with dependency to relaxed to only integration_runner binary.

With improved nr-php testing service following tests need fixing:
* memcached integration tests
* memcache integration tests for php 8.2
* pgsql integration tests

Following tests are disabled because their result depend on runtime environment:
* tests/integration/attributes/test_disabled_browser.php
* tests/integration/attributes/test_enabled_browser.php
* tests/integration/attributes/test_enabled_browser_old.php
* tests/integration/attributes/test_precedence_browser.php

Following tests are disabled because they fail:
* tests/integration/external/curl_multi_exec/test_curl_multi_exec_params.php

### CODE REVIEW

**Review by mfulb — APPROVED** [review]
Excellent work Michal!

File: .github/workflows/test-agent.yml [hunk 1]
```diff
@@ -143,3 +150,117 @@ jobs:
...
+        with:
+          name: newrelic.so-${{matrix.platform}}-${{matrix.arch}}-${{matrix.php}}
+          path: php-agent/agent/modules/newrelic.so
+  integration-tests:
+    needs: [daemon-unit-tests, agent-unit-test]
+    runs-on: ubuntu-latest
+    env:
+      IMAGE_NAME: newrelic/nr-php-agent-builder
+      IMAGE_TAG: make-php
+      IMAGE_VERSION: v1
```
[the remark this answers is not in the document]
Comment (lavarou) [reply .github/workflows/test-agent.yml:164]: Yes, there is - image version is going to be centralized. Versions will be stored as repository environmental variables and those will be used throughout workflows.

**Review by ZNeumann — APPROVED** [review]
Can't wait to have this in!

**Review by zsistla — APPROVED** [review]
Great job, thanks!

**Pull request merged**